src/inference/forwardSample.js

```diff
@@ -68,6 +70,12 @@ module.exports = function(env) {
     },
 
     factor: function(s, k, a, score) {
+      if (!this.opts.guide && !this.factorWarningIssued) {
```

Comment:
Makes sense.

There's an alternative view where specifying a guide only replaces the prior when sampling, and you're otherwise still doing inference as usual (but reweighting the samples to account for the fact that you're importance sampling in one form or another). For example, you could first fit a guide program and then use it instead of the prior as the basic proposal distribution in MCMC, which obviously wouldn't ignore factors. Your view is a better fit for how guide programs are currently used in WebPPL, but I think the alternative has merits as well, and if we went down that route, I think we'd want to issue warnings in `forward` even if a guide is specified.